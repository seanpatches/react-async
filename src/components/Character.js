import React from 'react';
import PropTypes from 'prop-types';

function Character({ character }){
  return (
    <>
    <p>{character.name}</p> 
    <p>{character.status}</p>
    <p>{character.age}</p>
    <p>{character.species}</p>
    </>
  ); 
}

Character.propTypes = {
  character: PropTypes.shape({
    name: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    age: PropTypes.string,
    species: PropTypes.string
  })
};

export default Character;
