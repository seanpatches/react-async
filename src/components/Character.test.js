import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders a Character', () => {
    const character = {
      name: 'Sean',
      status: 'Alive?',
      image: 'imaged',
      species: 'Human',
    };
    const wrapper = shallow(<Character character={character} />);
    expect(wrapper).toMatchSnapshot();
  });
});
