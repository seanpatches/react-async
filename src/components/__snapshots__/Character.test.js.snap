// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders a Character 1`] = `
<Fragment>
  <p>
    Sean
  </p>
  <p>
    Alive?
  </p>
  <p>
    27
  </p>
  <p>
    Human
  </p>
</Fragment>
`;
